space-concordia-robotics/robotics-prototype
# Custom msgs 297

# Assignee Section

## Description
Restored my custom message types after a tip from @MartensCedric turned out to solve the issue. Turns out that what probably happened in the past was I forgot to catkin_make on the rover when I was using the custom message types. To avoid this in the future I wrote a quick set of instructions on what to do for future custom message types.

- Restored the custom messages I created
- In messageType declarations in ros-helpers.js etc, specified the source package when not previously explicitly specified (e.g. `ArmRequest` becomes `mcu_control/ArmRequest`)
- Restored hidden navigation panels to the GUI (hidden during ERC)
- Had a successful test after pulling and running catkin_make
- Added a wiki guide to integrating new custom messages/etc to GUI
- Added flags to the custom message types to handle cases where data is unavailable
- Updated the GUI to handle cases where data is unavailable
- [ ] Still need to add a guide to running navigation stuff somewhere

List of revived custom messages:
- `/battery_voltage`
- `/battery_temps`
- `/rover_position`
- `/rover_goal`
- `/antenna_goal`

To Test:
1) Switch to this branch on your machine and on the odroid. `catkin_make` in `/rospackages` on both machines.
2) In two terminals, run the GUI with `rosgui` and `startgui`
3) In two new terminals, run the navigation nodes currently held in `mcu_control` called `AntennaNode.py` and `NavigationNode.py`. I believe they can also be run in the odroid, but the current expected use case is that they run on the basestation.
4) (local mode) Open s bunch of new terminals and use tab complete to publish test messages to the topics affected by this issue with `rostopic pub -r 2 <topicName> <messageType> <messageContent>`. Put nonzero values for best results.
4) (competition mode) SSH into the Odroid in two terminals and `rosrun mcu_control ArmNode.py usb` and `rosrun mcu_control RoverNode.py usb`. Verify on your computer and on the odroid with `rostopic info <messagName>` that the messages exist and are using the "new" custom message types.
5) Now that everything is set up, open the GUI and verify a few things. On all pages, the battery temperatures and voltage shouldn't be '-' anymore, they should have values. On the rover page, the navigation panel should be back up.
6) Input random data for the antenna values and gps goal. You should see all of the fields related to rover lat/long/heading and rover/antenna desired directions set to some value besides '-'.

closes #297 and #128 

The approval from all software team leads is necessary before merging.

# Reviewer Section

Aside from local testing and the General Integration Test it is implied that static analysis should be included in the verification process.


For Pull Requests that do not include code changes, it is not required to perform the tests above.
